src/rule.ts

```diff
@@ -3,26 +3,26 @@ import { JSONPathExpression } from 'nimma';
 
 import { IDocument } from './document';
 import { DEFAULT_SEVERITY_LEVEL, getDiagnosticSeverity } from './ruleset/severity';
-import { IGivenNode, IRule, IThen, SpectralDiagnosticSeverity } from './types';
+import { IGivenNode, IProcessedRule, IThen } from './types';
 import { hasIntersectingElement } from './utils';
+import { DiagnosticSeverity } from '@stoplight/types';
 
 export class Rule {
   public readonly name: string;
   public readonly description: string | null;
   public readonly message: string | null;
-  public readonly severity: SpectralDiagnosticSeverity;
+  public readonly severity: DiagnosticSeverity;
   public readonly resolved: boolean;
   public readonly formats: Optional<string[]>;
 
   public readonly then: IThen[];
   public readonly given: string[];
 
-  public get enabled(): boolean {
-    return this.severity !== -1;
-  }
+  public readonly enabled: boolean;
 
-  constructor(name: string, rule: IRule) {
+  constructor(name: string, rule: IProcessedRule) {
     this.name = name;
+    this.enabled = rule.enabled !== false;
```

Comment:
```suggestion
    this.enabled = rule.enabled ?? true;
```

seems more consistent with other rules, and IMO it's easier to read 🤷‍♂️